theano/configdefaults.py

```diff
@@ -42,6 +42,7 @@ def floatX_convert(s):
              "Note: float16 support is experimental, use at your own risk.",
              EnumStr('float64', 'float32', 'float16',
                      convert=floatX_convert,),
+             in_c_key=False
              )
```

Comment:
That flag was the whole reason for introducing `in_c_key` in the first place.
The issue was that some nodes, maybe Elemwise or CAReduce, would have an output dtype that depends not only on the input's dtypes, but also on the _current_ value of `floatX` (not the one when the node was built). It means that if C code is compiled or reloaded, and _then_ the value of `floatX` changes, then the value computed would have a different type from the output variable.